source/distributed/tria.cc

```diff
@@ -4171,6 +4171,37 @@ namespace parallel
 
 
 
+    template <int dim, int spacedim>
+    types::subdomain_id
+    Triangulation<dim, spacedim>::subdomain_id(const CellId &cell_id) const
+    {
+      // find descendent from coarse quadrant
+      typename dealii::internal::p4est::types<dim>::quadrant p4est_cell,
+        p4est_children[GeometryInfo<dim>::max_children_per_cell];
+
+      dealii::internal::p4est::init_coarse_quadrant<dim>(p4est_cell);
+      for (const auto &child_index : cell_id.get_child_indices())
+        {
+          dealii::internal::p4est::init_quadrant_children<dim>(p4est_cell,
+                                                               p4est_children);
+          p4est_cell = p4est_children[static_cast<unsigned int>(child_index)];
```

Comment:
Ah, I see that you already found how to do this with p4est. There must be a similar functionality already in this .cc file for other cases, right?

Reply:
Yes. I basically took inspiration from this section and preceded it with the creation of the p4est quadrant.
https://github.com/dealii/dealii/blob/a40844fdb863d99efc4561d21542417298a4a0a1/source/distributed/tria.cc#L398-L405

Reply:
We might think of going through the p4est structure recursively only once to determine just the subdomains of cells we're interested in. This might speed things up a bit.

Reply:
I think that the call to `p4est_children` doesn't actually access any data structure. Note how the call doesn't get a pointer to a tree or other p4est data structure: It really just copies the quadrant 4 (or 8) times and then sets a few bits. It must really be quite cheap.